File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var reverseDegree = function(s) {
    let result = 0;

    for (let i = 0; i < s.length; i++) {
        const reverseValue = 26 - (s.charCodeAt(i) - 97); // 97 is the ASCII code for 'a'
        result += reverseValue * (i + 1);
    }

    return result;
};

File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.c
#include <string.h>

int reverseDegree(char* s) {
    int result = 0;
    int len = strlen(s);

    for (int i = 0; i < len; i++) {
        int reverseValue = 26 - (s[i] - 'a');
        result += reverseValue * (i + 1);
    }

    return result;
}
